impeller/renderer/backend/vulkan/resource_manager_vk_unittests.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <sys/types.h>
+#include <functional>
+#include <memory>
+#include <utility>
+#include "gtest/gtest.h"
+#include "impeller/renderer/backend/vulkan/resource_manager_vk.h"
+
+namespace impeller {
+namespace testing {
+
+// While expected to be a singleton per context, the class does not enforce it.
+TEST(ResourceManagerVKTest, CreatesANewInstance) {
+  auto const a = ResourceManagerVK::Create();
+  auto const b = ResourceManagerVK::Create();
+  EXPECT_NE(a, b);
+}
+
+// Invokes the provided callback when the destructor is called.
+//
+// Can be moved, but not copied.
+class DeathRattle final {
```

Comment:
Put this in an anonymous namespace to make sure it doesn't collide.  Anything that creates symbols should be in a header or in an anonymous namespace (modulo the weird stuff googletest is doing).

Reply:
Ah thanks, done!